src/hotspot/cpu/x86/x86.ad

```diff
@@ -5348,6 +5406,18 @@ instruct vaddD_mem(vec dst, vec src, memory mem) %{
   ins_pipe( pipe_slow );
 %}
 
+// Halffloat vector add
+instruct vaddHF_reg(vec dst, vec src1, vec src2) %{
+  predicate(UseAVX > 2);
```

Comment:
We can check existence of AVX3 targets match_rule_supported for half float IR nodes and remove explicit predicates from each pattern.